### DISCUSSION TITLE
Variables in Python Comment illegal terms below Answers

### DISCUSSION THREAD

**Discussion by Abuthahir94** [community/community#29642] [Other Feature Feedback, Questions, & Ideas]
1)  Which of the following variable names are illegal in Python?  (Select three answers)
           1) my_var

           2) n

           3) 205

           4) averylongVariablename

          5) n205

          6) n 205

          7) Del

          8) del

Kindly Suggest the Correct 3 Answers and How to do you say the explanation in these questions.

**Comment by Variable-py** [comment]
3 & 6 & 8

**Reply by Abuthahir94** [reply]
Thanks Mr.Variable-Py first of all, if any clarification about the illegal Variables please kindly describe the illusion.

**Reply by venuslatte** [reply]
Can you please help me?

**Reply by Abuthahir94** [reply]
It's pleasure what you want to me mam?

**Comment by DenverCoder1** [comment]
Hopefully this helps

* Variables must start with either a letter or underscore, they **can't** start with numbers
eg. `abc` and `abc1` are allowed, but `1abc` and `205` are **not** allowed.
* The remaining characters must all be letters, numbers, or underscores, you **can't** have spaces, hyphens, etc.
eg. `abc_123d` and `_205abc` allowed, but `n 205` and `abc-123` are **not** allowed.
* Built-in Python keywords are not allowed to be used as variable names since they have other meanings (see https://www.w3schools.com/python/python_ref_keywords.asp), also note that all terms and keywords are **case-sensitive**.
eg. `del`, `yield`, `return`, etc (see list) are **not** allowed, but with slight modifications (eg. `Del`, `_yield`, etc.) they would be legal (although I wouldn't recommended it).

**Reply by Abuthahir94** [reply]
Thanks for your kind clarification pal.

**Comment by levintech** [comment]
There are rules for naming Python variables:

- A variable name must start with a letter or the underscore character
- A variable name cannot start with a number
- A variable name can only contain alpha-numeric characters and underscores (A-z, 0-9, and _ )
- Variable names are case-sensitive (age, Age and AGE are three different variables)
- Variable names shouldn't be same with keywords

Followings are illegal in Python

- **205**
> A variable name cannot start with a number

- **n 205**

> A variable name can only contain alpha-numeric characters and underscores (A-z, 0-9, and _ )
> But this variable name contain space character. " "

 
- **del**

> Variable names shouldn't be same with keywords
> `del` is a Python keyword
